server/src/test/java/org/apache/druid/server/ClientQuerySegmentWalkerTest.java

```diff
@@ -798,7 +798,36 @@ public void testTimeseriesOnGroupByOnTableErrorTooManyRows()
                                 .withId(DUMMY_QUERY_ID);
 
     expectedException.expect(ResourceLimitExceededException.class);
-    expectedException.expectMessage("Subquery generated results beyond maximum[2]");
+    expectedException.expectMessage("Subquery generated results beyond maximum[2] rows");
+
+    testQuery(query, ImmutableList.of(), ImmutableList.of());
+  }
+
+
+  @Test
+  public void testTimeseriesOnGroupByOnTableErrorTooLarge()
```

Comment:
I think tests around
1. String cols
2. Long cols
3. Double/float cols
4. Complex cols
5. Array cols
6. Nested cols 
Would help us build confidence in the feature. 

Some tests to checkout: 
* `CalciteQueryTests#testMaxSubqueryRows`
* `GroupByQueryRunnerTest#testGroupByMaxRowsLimitContextOverride`

Ideally all tests which have a subquery should be executed using the new code path but since its feature flagged it might not be a hard requirement.